### DISCUSSION TITLE
[@types/react] What should we do with React.Element?

### DISCUSSION THREAD

**Discussion by mattpocock** [DefinitelyTyped/DefinitelyTyped#68458] [Issues with a @types package]
`React.Element` is an odd type. It essentially acts as an alias to React.JSX.Element, except for in the cases where it's used with `createElement`, where it preserves the props passed in.

Should we be recommending that users mostly use `React.JSX.Element` instead? What should I write for the JSDoc comments on it?

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
`React.JSX.Element` acts as `ReactElement` not the other way around.

The existence of the JSX namespace is a TypeScript limitation. Ideally `React.JSX` would not exist. The recommendation is to use the same type that's in `ReactNode`: `ReactElement`. 

The plan is also to get rid of the prefix so that it's just `React.Node` or `React.Element`.
